File: backend/app/schemas/record.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.record import RecordStatus


class RecordRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    title: str
    original_filename: str
    file_type: str
    file_size: int
    status: RecordStatus
    transcript: str | None
    summary: str | None
    error_message: str | None
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/record.ts
export type RecordStatus =
  | "uploaded"
  | "transcribing"
  | "transcribed"
  | "summarizing"
  | "completed"
  | "failed";

export interface RecordItem {
  id: string;
  title: string;
  original_filename: string;
  file_type: string;
  file_size: number;
  status: RecordStatus;
  transcript: string | null;
  summary: string | null;
  error_message: string | null;
  created_at: string;
  updated_at: string;
}
